jcs/openst.py

```diff
@@ -36,12 +36,12 @@ def os_instance_create(self, instance_name, instance_type, image_name, key_name,
         if not instance_type:
             raise Exception('OS instance type (flavor) "{}" not found'.format(instance_type))
 
-        network_fixed = self._conn.get_network(network_fixed)
+        network_fixed = self._conn.get_network(network_fixed) or network_public
```

Comment:
Hm... this is kinda mix of value types, while network_fixed is lookup, the network_public remains just a string. I guess create_server can deal with that. But maybe it is better to lookup the public network also and report user if none of the network are found?

Reply:
ah.... and there is another thing.... look later how public network is used for making a floating ip. This code mostly used for our internal openstack cloud where we have floating ips, while OVH does not provide floating IP mechanism.